src/main/kotlin/example/SidebarOpenedModule.kt

```diff
@@ -1,21 +1,29 @@
 package example
 
+import mui.system.ThemeProvider
 import react.*
 
 data class SidebarOpened(
     val value: Boolean,
     val setValue: StateSetter<Boolean>,
+    val darkMode: Boolean,
+    val setDarkMode: StateSetter<Boolean>
 )
 
 val SidebarOpenedContext = createContext<SidebarOpened>()
 
 val SidebarOpenedModule = FC<PropsWithChildren> { props ->
+    val (darkMode, setDarkMode) = useState(false)
     val (open, setOpen) = useState(true)
-    val sidebarOpened = useMemo(open, setOpen) {
-        SidebarOpened(open, setOpen)
+    val sidebarOpened = useMemo(open, setOpen, darkMode, setDarkMode) {
+        SidebarOpened(open, setOpen, darkMode, setDarkMode)
     }
 
     SidebarOpenedContext.Provider(sidebarOpened) {
-        props.children()
+        val newTheme = if (darkMode) themeDark else themeLight
+        ThemeProvider {
```

Comment:
`ThemeProvider` shouldn't be used in `SidebarOpenedModule` component.
Please check the examples from https://mui.com/customization/theming/#api